File: Products/sample-android/app/src/main/java/mockifer/example/android/common/Requester.java
package mockifer.example.android.common;

import okhttp3.Callback;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;

public class Requester {
    public static final String BASE_URL = "http://localhost:8503";
    private static OkHttpClient client = new OkHttpClient();

    private Requester() {

    }

    public static void doRequest(String method, String uri, String body, Callback callback) {
        RequestBody requestBody = null;

        if (body != null) {
            requestBody = RequestBody.create(MediaType.parse("utf-8"), body);
        }

        Request request = new Request.Builder()
                .url(BASE_URL + uri)
                .method(method, requestBody)
                .build();

        client.newCall(request).enqueue(callback);
    }
}

File: Support/samples/sample-android/app/src/main/java/mockifer/example/android/common/Requester.kt
package mockifer.example.android.common

import mockifer.example.android.BuildConfig
import okhttp3.*

object Requester {
    private val client by lazy { OkHttpClient() }

    fun doRequest(method: String, uri: String, body: String?, callback: Callback) {
        val requestBody = body?.let {
            RequestBody.create(MediaType.parse("utf-8"), it)
        }

        val request = Request.Builder()
                .url(BuildConfig.SERVER_BASE_URL + uri)
                .method(method, requestBody)
                .build()

        client.newCall(request).enqueue(callback)
    }
}
